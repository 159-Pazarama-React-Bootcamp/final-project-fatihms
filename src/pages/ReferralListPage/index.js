import React from "react";

import { Link } from "react-router-dom";

import { BiExit } from "react-icons/bi";

import Header from "../../components/Header";
import ReferralList from "../../containers/ReferralList";

import styles from "./styles.module.css";

function ReferralListPage() {
  return (
    <div className={styles["referral-list-container"]}>
      <header>
        <Header title="BaÅŸvuru Listesi">
          <Link to="/admin" className={styles["mac-link"]}>
            <i>
              <BiExit size={30} color="white" />
            </i>
          </Link>
        </Header>
      </header>
      <main>
        <article>
          <ReferralList />
        </article>
      </main>
    </div>
  );
}

export default ReferralListPage;
