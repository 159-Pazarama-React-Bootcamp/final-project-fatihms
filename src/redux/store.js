import { configureStore } from "@reduxjs/toolkit";

import codeReducer from "./code/codeSlice";

const store = configureStore({
  reducer: { code: codeReducer },
});

export default store;
