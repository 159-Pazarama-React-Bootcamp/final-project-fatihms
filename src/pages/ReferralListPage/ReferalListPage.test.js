/*eslint-disable */

import React from "react";

import { render, screen } from "@testing-library/react";
import renderer from "react-test-renderer";
import { MemoryRouter as Router } from "react-router-dom";
import userEvent from "@testing-library/user-event";
import "@testing-library/jest-dom/extend-expect";

import { Provider } from "react-redux";
import store from "../../redux/store";

import ReferralListPage from "./index";

describe("ReferralListPage", () => {
  beforeEach(() => {
    render(
      <Provider store={store}>
        <Router>
          <ReferralListPage />
        </Router>
      </Provider>
    );
  });
  test("ReferralListPage renders correctly", () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <Router>
            <ReferralListPage />
          </Router>
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
