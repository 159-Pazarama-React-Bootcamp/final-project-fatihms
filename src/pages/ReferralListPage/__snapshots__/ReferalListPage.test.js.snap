// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReferralListPage ReferralListPage renders correctly 1`] = `
<div
  className="referral-list-container"
>
  <header>
    <div
      className="header-container"
    >
      <h2>
        Ba≈üvuru Listesi
      </h2>
      <a
        className="mac-link"
        href="/admin"
        onClick={[Function]}
      >
        <i>
          <svg
            color="white"
            fill="currentColor"
            height={30}
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": "white",
              }
            }
            viewBox="0 0 24 24"
            width={30}
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M19.002 3h-14c-1.103 0-2 .897-2 2v4h2V5h14v14h-14v-4h-2v4c0 1.103.897 2 2 2h14c1.103 0 2-.897 2-2V5c0-1.103-.898-2-2-2z"
            />
            <path
              d="m11 16 5-4-5-4v3.001H3v2h8z"
            />
          </svg>
        </i>
      </a>
    </div>
  </header>
  <main>
    <article>
      <div
        className="loading"
      >
        <div
          className="lds-ring"
        >
          <div>
             
          </div>
          <div>
             
          </div>
          <div>
             
          </div>
          <div>
             
          </div>
        </div>
      </div>
    </article>
  </main>
</div>
`;
